src/main/java/org/breedinginsight/utilities/response/ResponseUtils.java

```diff
@@ -0,0 +1,214 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.breedinginsight.utilities.response;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.exceptions.HttpStatusException;
+import org.apache.commons.lang3.tuple.Pair;
+import org.breedinginsight.api.model.v1.request.query.QueryParams;
+import org.breedinginsight.api.model.v1.request.query.SearchRequest;
+import org.breedinginsight.api.model.v1.response.DataResponse;
+import org.breedinginsight.api.model.v1.response.Response;
+import org.breedinginsight.api.model.v1.response.metadata.Metadata;
+import org.breedinginsight.api.model.v1.response.metadata.Pagination;
+import org.breedinginsight.api.model.v1.response.metadata.Status;
+import org.breedinginsight.api.model.v1.response.metadata.StatusCode;
+import org.breedinginsight.api.v1.controller.metadata.SortOrder;
+import org.breedinginsight.model.ProgramUpload;
+import org.breedinginsight.utilities.response.mappers.AbstractQueryMapper;
+import org.breedinginsight.utilities.response.mappers.FilterField;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ResponseUtils {
+
+    public final static Integer DEFAULT_PAGE_SIZE = 50;
+    public final static Integer DEFAULT_PAGE = 1;
+    public final static SortOrder DEFAULT_SORT_ORDER = SortOrder.ASC;
+
+    // All
+    public static <T> HttpResponse<Response<DataResponse<T>>> getQueryResponse(
+            List data, AbstractQueryMapper mapper, SearchRequest searchRequest, QueryParams queryParams) {
+        return processSearchResponse(data, searchRequest, queryParams, mapper, new Metadata());
+    }
+
+    // Pagination and sort only
+    public static <T> HttpResponse<Response<DataResponse<T>>> getQueryResponse(
+            List data, AbstractQueryMapper mapper, QueryParams queryParams) {
+        return processSearchResponse(data, null, queryParams, mapper, new Metadata());
+    }
+
+    // All
+    public static <T> HttpResponse<Response<ProgramUpload>> getQueryResponse(
+            ProgramUpload upload, AbstractQueryMapper mapper, SearchRequest searchRequest, QueryParams queryParams) {
+        return processSearchResponse(upload, searchRequest, queryParams, mapper, new Metadata());
+    }
+
+    // Pagination and sort only
+    public static <T> HttpResponse<Response<ProgramUpload>> getQueryResponse(
+            ProgramUpload upload, AbstractQueryMapper mapper, QueryParams queryParams) {
+        return processSearchResponse(upload, null, queryParams, mapper, new Metadata());
+    }
```

Comment:
I think it'd be helpful to have the names of these methods reflect that they are for uploads instead of overloading the method name of `getQueryResponse`

Reply:
How so? I plan on changing the names, but still wondering your thoughts as to the benefits of changing the method name instead of overloading?

Reply:
Changed and pushed

Reply:
My main reason for saying that is because of this `HttpResponse<Response<ProgramUpload>>` being specifically typed to `ProgramUpload`, and the overloaded method has it as generice `<T>`

Reply:
Created new methods for ProgramUpload and removed the generic typing from those methods.